GPS.Data/Parser.cs

```diff
@@ -1,101 +1,160 @@
-            Serializer = new XmlSerializer(typeof(GpsIsFile));
-            using Stream reader = new FileStream(filePath, FileMode.Open);
-            Outages = (GpsIsFile)Serializer.Deserialize(reader);
-        }
+		private readonly List<string> _errorLog = new List<string>();
+		/// <summary>
+		/// A list of strings containing all the errors that arise from the validation of the .sof file.
+		/// </summary>
+		public List<string> ErrorLog { get { return _errorLog; } }
 
         /// <summary>
-        /// Takes a .sof and populates the classes with the information included therein
+        /// object in reference to Validate.cs
         /// </summary>
-        /// <returns>
-        /// A list of all outages consolidated in one place for ease of use when querying
-        /// </returns>
-        public List<Outage> PopulateObjectsFromSof()
-        {
-            //Populate the AllOutages List
-            List<Outage> allOutages = new List<Outage>();
-            foreach (Historical historicalOutage in Outages.HistoricalOutages)
-            {
-                //Add validation here!
+        readonly Validate ValidateObj = new Validate();
+		/// <summary>
+		/// Object Created to coordinate with the lock
+		/// </summary>
+		private readonly object SerializeLock = new object();
 
-                allOutages.Add(new Outage
+		public Parser(string filePath)
+		{
+			try
+			{
+				//Thread protection for whatever file is passed into the serializer
+				lock (SerializeLock)
                 {
-                    TagName = "HISTORICAL",
-                    SatelliteVehicleId = historicalOutage.SatelliteVehicleId,
-                    SatelliteVehicleNumber = historicalOutage.SatelliteVehicleNumber,
-                    Name = historicalOutage.Name,
-                    Type = historicalOutage.Type,
-                    Reference = historicalOutage.Reference,
-                    StartYear = historicalOutage.StartYear,
-                    StartDayOfYear = historicalOutage.StartDayOfYear,
-                    StartHour = historicalOutage.StartHour,
-                    StartMinute = historicalOutage.StartMinute,
-                    StartSecond = historicalOutage.StartSecond,
-                    EndYear = historicalOutage.EndYear,
-                    EndDayOfYear = historicalOutage.EndDayOfYear,
-                    EndHour = historicalOutage.EndHour,
-                    EndMinute = historicalOutage.EndMinute,
-                    EndSecond = historicalOutage.EndSecond
-                });
-            }
+					//Serializes the .sof and populates each of the classes
+					Serializer = new XmlSerializer(typeof(GpsIsFile));
+					using Stream reader = new FileStream(filePath, FileMode.Open);
+					Outages = (GpsIsFile)Serializer.Deserialize(reader);
+				}
+			}
+			catch (FileNotFoundException)
+			{
+				Console.WriteLine("FileNotFoundException");
+				//TODO: navigate to an error page if this is the case
+				System.Environment.Exit(1);
```

Comment:
This will be a class used outside of test program.  Allowing it to throw an exception is good enough (used exit at the top level where you within the application itself)